Makefile

```diff
@@ -134,6 +134,32 @@ enter_%:
 		-w $(PROGNOSTIC_RUN_WORKDIR) \
 		$(REGISTRY)/$*:$(VERSION) bash
 
+SCREAM_INSTALL_PATH ?= docker/prognostic_scream_run/scream
+SCREAM_BRANCH ?= master
+clone_scream_repository:
```

Comment:
This doesn't work because clone calls update, if update depends on clone, then it's an infinite loop.
It is indeed a bit awkward here, the goal is to clone scream repo and submodule update. It should never hit `SCREAM_INSTALL_PATH` already exist on CI since it's a fresh build each time, but when developing locally, I have the repo cloned already, so I opted to just update the branch whenever clone gets called again. This may not be an expected behavior either. 
The other option is to delete `update_scream_repository` altogether, and throw an error whenever `SCREAM_INSTALL_PATH` already exists. Assuming the user know what they're doing, we can allow bypassing the error and user is responsible for updating the branch.

Reply:
As discussed, the ask here is just to move the call of `make update_scream_repository` out of `clone_scream_repository` and instead do it as a makefile prerequisite.